cirq/protocols/resolve_parameters_test.py

```diff
@@ -143,3 +143,22 @@ def test_recursive_resolve():
     assert cirq.resolve_parameters_once(a, resolver) == b
     with pytest.raises(RecursionError):
         _ = cirq.resolve_parameters(a, resolver)
+
+
+# TODO: remove in Cirq v0.11
+def test_backwards_compatible():
```

Comment:
In accordance with my other comment I'd add another test that would cover the case when the `TypeError` is not due to the specific incompatibility that we're trying to fix here.

Reply:
Opted to avoid `try/except` altogether - see Doug's comment.